app/geant-exporter/geant-exporter.cc

```diff
@@ -0,0 +1,246 @@
+        is_stable = particle->GetPDGStable();
+
+        // Converting lifetime of unstable particles to seconds
+        // For stable particles lifetime = -1
+        if (!is_stable)
+        {
+            lifetime *= 1E-9;
+        }
+
+        // ROOT does not like special characters for its TTree names
+        replace_characters(name, "+", "Plus");
+        replace_characters(name, "-", "Minus");
+
+        // Gamma is a ROOT function. A TTree with that same name confuses ROOT
+        if (name == "gamma")
+        {
+            name = "photon";
+        }
+
+        // Creating a TTree for each particle
+        TTree* tree_particle = new TTree(name.c_str(), name.c_str());
+
+        // Preserving the original particle name in the data
+        name = particle->GetParticleName();
+
+        tree_particle->Branch("name", &name);
+        tree_particle->Branch("pdg", &pdg, "pdg/I");
+        tree_particle->Branch("mass", &mass, "mass/D");
+        tree_particle->Branch("charge", &charge, "charge/D");
+        tree_particle->Branch("spin", &spin, "spin/D");
+        tree_particle->Branch("lifetime", &lifetime, "lifetime/D");
+        tree_particle->Branch("is_stable", &is_stable, "is_stable/O");
+
+        tree_particle->Fill();
+
+        std::cout << "  Added " << name << std::endl;
+    }
+
+    root_output->Write();
+    root_output->Close();
+    std::cout << root_output_filename << " done!" << std::endl;
+    std::cout << std::endl;
+
+    //-----------------------------------------------------------------------//
+    /*!
+     * LOOP 2: Storing physics tables into a ROOT file
+     */
+    //-----------------------------------------------------------------------//
+
+    GeantPhysicsTableParser table_parser;
+    table_parser.root_file_name(root_output_filename);
+
+    particle_iterator->reset();
+
+    while ((*particle_iterator)())
+    {
+        // Fetching particle and its data
+        G4ParticleDefinition* particle = particle_iterator->value();
+
+        // See comments in LOOP 1 on why we are skipping the geantino
+        if (particle->GetParticleName() == "geantino")
+            continue;
+
+        std::cout << "=============" << std::endl;
+        std::cout << particle->GetParticleName() << std::endl;
+        std::cout << "=============" << std::endl;
+
+        G4ProcessVector* process_list
+            = (particle->GetProcessManager())->GetProcessList();
+
+        // Looping over processes
+        for (std::size_t j = 0; j < process_list->size(); j++)
+        {
+            G4VProcess* process = (*process_list)[j];
+            table_parser.add_physics_table(process, particle);
+        }
+    }
+
+    std::cout << root_output_filename << " done!" << std::endl;
+
+    //---------------------------- Job termination --------------------------//
+    /*!
+     * User actions, physics list, and detector construction are owned and
+     * deleted by the run manager
+     */
+    delete run_manager;
```

Comment:
Sure, I can do that. I never considered using a unique_ptr here simply because the run manager is always created at the beginning and deleted at the end of the main scope, so a wrapper around a raw pointer seemed unnecessary.

Also, it is one of those habits. Most HEP codes are written with raw pointers and you follow suit.

Reply:
No worries. So in this case I think the right choice is just make the variable local `G4RunManager run_manager;` since there's no need to manually memory manage (the language's built-in scope management will automatically call the destructor this way).